src/policy/rbf.cpp

```diff
@@ -56,7 +56,7 @@ RBFTransactionState IsRBFOptInEmptyMempool(const CTransaction& tx)
 }
 
 std::optional<std::string> GetEntriesForConflicts(const CTransaction& tx,
-                                                  CTxMemPool& pool,
+                                                  const CTxMemPool& pool,
```

Comment:
`&` doesn't create a copy of CTxMemPool, so I think the pull title is wrong.

Also, I wonder if this is something that can be enforced with clang-tidy, if we want it. Maybe https://clang.llvm.org/extra/clang-tidy/checks/readability/non-const-parameter.html ?

Reply:
Thanks Marco, updated the title and commit message and testing your clang-tidy idea.

Reply:
Dropped the tidy check as it proposed making a pointer to const value that needs to be mutable.

Reply:
Yeah, I don't know if there is a tidy check for this. But when the reason for the changes in this pull are "consistency" https://github.com/bitcoin/bitcoin/pull/23076#issuecomment-930017016 , we should enforce this "consistency" to avoid incremental fixups to existing code.

Same for the CFeeRate change.

Reply:
Many changes seem to be based on copying or grepping the existing code. For example, if CFeeRate in-params are all passed in the same fashion rather than, say, 2/3 by const ref and 1/3 by value, ISTM that alone may help to maintain consistency.

Reply:
How do you prevent a change from introducing this tomorrow? And then a follow-up pull to that change in tomorrow+n days?

Reply:
Removed consistency from the pull description summary (I think there is ample remaining rationale).

Reply:
My comment was meant generally, I only filled in a random rationale as an example.

> When the reason for the changes in this pull are "$rationale", we should ideally enforce this "$rationale" to avoid incremental fixups to existing or future code.

Reply:
> How do you prevent a change from introducing this tomorrow? And then a follow-up pull to that change in tomorrow+n days?

This is true for many changes that are merged.

There was plenty of rationale provided in the PR description and it seems pointless to continue to spend more time here.